### PR TITLE
feat(component): add support for watchPosition()

### PR DESCRIPTION
**Pull request by codeocelot**
Was bored this afternoon, implemented the feature I requested in #170.  I did my best to follow your contribution guideline: please let me know if anything's amiss and I'll happily fix it.  

Add support for geolocation.watchPosition() method, updating state on each location change and
remove watcher on component unmount.

impl #170

### CODE REVIEW

File: src/components/geolocated.js [hunk 1]
```diff
@@ -64,17 +65,25 @@ const geolocated = ({
...
 
         getLocation() {
-            if (!geolocationProvider || !geolocationProvider.getCurrentPosition) {
+            if (!geolocationProvider || !geolocationProvider.getCurrentPosition || !geolocationProvider.watchPosition) {
                 throw new Error('The provided geolocation provider is invalid');
             }
 
+            const funcPosition = (watchPosition
+              ? geolocationProvider.watchPosition
+              : geolocationProvider.getCurrentPosition).bind(geolocationProvider);
```
Comment (codeocelot) [inline src/components/geolocated.js]: looks weird, but `getCurrentPosition/watchPosition` require a geolocation context to work.

**Comment by no23reason** [comment]
Hi, thank you for your contribution! Sorry I took so long to look at it, I was ill :-/ It looks good to me, could you please also update the README with docs of the new option?

**Comment by codeocelot** [comment]
Great catch, thanks for noticing that, updated README with details on the new prop.  Hope you're feeling better!

**Comment by no23reason** [comment]
Thank you and thanks for the update, this is great! :)

**Pull request merged**